tests/compute.py

```diff
@@ -359,3 +360,70 @@ def compute_image_load_store(cfg: SceneCfg, show_dbg_points=False):
         group.add_children(get_debug_points(cfg, cuepoints))
 
     return group
+
+
+_IMAGE_3D_STORE_COMPUTE = """
+void main()
+{
+    ivec2 pos = ivec2(gl_LocalInvocationID.xy);
+    float size = float(imageSize(texture).x);
+    float color = (float(pos.x) + float(pos.x * pos.y)) / (size * size) ;
+    imageStore(texture, ivec3(pos, 0), vec4(color));
+    imageStore(texture, ivec3(pos, 1), vec4(color));
+    imageStore(texture, ivec3(pos, 2), vec4(color));
+}
+"""
+
+_IMAGE_3D_LOAD_STORE_COMPUTE = """
+void main()
+{
+    ivec2 pos = ivec2(gl_LocalInvocationID.xy);
+    vec4 color;
+    color.r = imageLoad(texture, ivec3(pos, 0)).r;
+    color.g = imageLoad(texture, ivec3(pos, 1)).r;
+    color.b = imageLoad(texture, ivec3(pos, 2)).r;
+    color.a = 1.0;
+    color.rgb = (color.rgb * scale.factors.x) + scale.factors.y;
+    imageStore(texture_rgba, pos, color);
+}
+"""
+
+
+@test_cuepoints(points=_get_compute_histogram_cuepoints(), tolerance=1)
+@scene(show_dbg_points=scene.Bool())
+def compute_image_3d_load_store(cfg: SceneCfg, show_dbg_points=False):
+    size = _N
+    scale = ngl.Block(
+        fields=[ngl.UniformVec2(value=(-1.0, 1.0), label="factors")],
+        layout="std140",
+    )
```

Comment:
1. Can you add a comment explaining why the convoluted way of passing down these factors? (I'm assuming to test misc conflicts between block and texture bindings or something)
2. Can you also mention that this is causing a color inversion? Eventually in a comment above the 2nd compute "Load colors, invert them and store them back"
3. This scale could be moved closer to the 2nd pass compute

Reply:
Done.